From 31d8f24ab21dd9ec5e09ea4d54ccb99088515957 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 5 Dec 2022 12:10:36 +0100
Subject: [PATCH 0389/1695] drivers: stm32_rng: add RNG_NIST_CONFIG_C custom
 configuration

Adds RNG_NIST_CONFIG_C configuration and correct RNG_NIST_CONFIG_A whose
value was incorrect.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git aa971b5cca4c83f0cfe42bbfcb55dda075908bd3 drivers: stm32_rng: add RNG_NIST_CONFIG_C custom configuration]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I3a6befb2331df21717ca0fce515d2081979c3926
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/297210
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_rng.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index 07688333e..b8555dcb3 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -49,8 +49,9 @@
 
 #define RNG_FIFO_BYTE_DEPTH	U(16)
 
-#define RNG_NIST_CONFIG_A	U(0x0F00D00)
+#define RNG_NIST_CONFIG_A	U(0x0F00F00)
 #define RNG_NIST_CONFIG_B	U(0x1801000)
+#define RNG_NIST_CONFIG_C	U(0x0F00D00)
 #define RNG_NIST_CONFIG_MASK	GENMASK_32(25, 8)
 
 #define RNG_MAX_NOISE_CLK_FREQ	U(3000000)
-- 
2.47.2

